src/profile.jl

```diff
@@ -457,33 +464,22 @@ function capture(cfg)
     return (; host=host_trace, device=device_trace, nvtx=nvtx_trace)
 end
 
-"""
-    Profile.print([io::IO], data; kwargs...)
-
-Prints profiling results to `io`
-
-See also: [`@profile`](@ref), [`@profiled`](@ref)
-"""
-function print end
-
-print(data; kwargs...) =
-    print(stdout isa Base.TTY ? IOContext(stdout, :limit => true) : stdout, data; kwargs...)
-function print(io::IO, data; trace=false, raw=false)
-    data = deepcopy(data)
+function Base.show(io::IO, results::ProfileResults)
```

Comment:
Now it will be a long day before I understand our show methods hierarchy, but should this be?


```suggestion
function Base.show(io::IO, ::MIME"text/plain", results::ProfileResults)
```

Reply:
I think we don't, in this case. Basically, you want the ProfileResults to be rendered, almost always. So when doing `display`, which IIUC is `MIME"text/plain"`, but also when doing `string` or `print` (which I think doesn't use a MIME input?), or when rendering in Pluto (which, again IIUC, uses `text/html`).

I'm probably misunderstanding though, as I'm not very familiar with the show method hierarchy either, but I did verify the above cases and the output is rendered correctly.